modules/core/src/main/java/org/apache/ignite/internal/util/HashUtils.java

```diff
@@ -120,12 +105,50 @@ public static int hash32(int data, int seed) {
      * @param data The input long value.
      * @return The 32-bit hash.
      */
-    public static int hash32(long data, int seed) {
-        long hash = hash64(data, seed);
+    public static int hash32(long data) {
+        long hash = hash64(data, 0);
 
         return (int) (hash ^ (hash >>> 32));
     }
 
+    /**
+     * Combines two hash values.
+     *
+     * @param hash1 The first hash.
+     * @param hash2 The second hash.
+     * @return The combined 32-bit hash.
+     */
+    public static int combine(int hash1, int hash2) {
+        long hash = hash64(hash1, hash2);
+
+        return (int) (hash ^ (hash >>> 32));
```

Comment:
Are there any reasons why it is implemented this way?
Also, does `combine(a,b) == combine(b,a)`?
More descriptive javadoc probably is needed here.

Reply:
Refactored, added comments.